pimd/pim6_cmd.c

```diff
@@ -0,0 +1,244 @@
+
+	snprintf(pim_if_xpath, sizeof(pim_if_xpath), "%s/frr-pim:pim",
+		 VTY_CURR_XPATH);
+
+	pim_enable_dnode = yang_dnode_getf(vty->candidate_config->dnode,
+					   "%s/pim-enable", pim_if_xpath);
+	if (!pim_enable_dnode) {
+		nb_cli_enqueue_change(vty, pim_if_xpath, NB_OP_DESTROY, NULL);
+		nb_cli_enqueue_change(vty, ".", NB_OP_DESTROY, NULL);
+	} else {
+		if (!yang_dnode_get_bool(pim_enable_dnode, ".")) {
+			nb_cli_enqueue_change(vty, pim_if_xpath, NB_OP_DESTROY,
+					      NULL);
+			nb_cli_enqueue_change(vty, ".", NB_OP_DESTROY, NULL);
+		} else
+			nb_cli_enqueue_change(vty, "./enable", NB_OP_MODIFY,
+					      "false");
+	}
+
+	return nb_cli_apply_changes(vty, FRR_GMP_INTERFACE_XPATH,
+				    "frr-routing:ipv6");
+}
+
+DEFUN(interface_ipv6_mld_join, interface_ipv6_mld_join_cmd,
+      "ipv6 mld join X:X::X:X [X:X::X:X]",
+      IPV6_STR
+      IFACE_MLD_STR
+      "MLD join multicast group\n"
+      "Multicast group address\n"
+      "Source address\n")
+{
+	int idx_group = 3;
+	int idx_source = 4;
+	const char *source_str;
+	char xpath[XPATH_MAXLEN];
+	struct prefix_ipv6 p;
+
+	if (argc == 5) {
+		source_str = argv[idx_source]->arg;
+
+		(void)str2prefix_ipv6(source_str, &p);
+
+		if (IPV6_ADDR_SAME(&p.prefix, &in6addr_any)) {
+			vty_out(vty, "Bad source address %s\n",
+				argv[idx_source]->arg);
+			return CMD_WARNING_CONFIG_FAILED;
+		}
+	} else
+		source_str = "::";
+
+	snprintf(xpath, sizeof(xpath), FRR_GMP_JOIN_XPATH, "frr-routing:ipv6",
+		 argv[idx_group]->arg, source_str);
+
+	nb_cli_enqueue_change(vty, xpath, NB_OP_CREATE, NULL);
+
+	return nb_cli_apply_changes(vty, NULL);
+}
+
+DEFUN(interface_no_ipv6_mld_join, interface_no_ipv6_mld_join_cmd,
+      "no ipv6 mld join X:X::X:X [X:X::X:X]",
+      NO_STR
+      IPV6_STR
+      IFACE_MLD_STR
+      "MLD join multicast group\n"
+      "Multicast group address\n"
+      "Source address\n")
+{
+	int idx_group = 4;
+	int idx_source = 5;
+	const char *source_str;
+	char xpath[XPATH_MAXLEN];
+	struct prefix_ipv6 p;
+
+	if (argc == 6) {
+		source_str = argv[idx_source]->arg;
+
+		(void)str2prefix_ipv6(source_str, &p);
+
+		if (IPV6_ADDR_SAME(&p.prefix, &in6addr_any)) {
+			vty_out(vty, "Bad source address %s\n",
+				argv[idx_source]->arg);
+			return CMD_WARNING_CONFIG_FAILED;
+		}
+	} else
+		source_str = "::";
+
+	snprintf(xpath, sizeof(xpath), FRR_GMP_JOIN_XPATH, "frr-routing:ipv6",
+		 argv[idx_group]->arg, source_str);
+
+	nb_cli_enqueue_change(vty, xpath, NB_OP_DESTROY, NULL);
+
+	return nb_cli_apply_changes(vty, NULL);
+}
+
+DEFUN(interface_ipv6_mld_version,
+      interface_ipv6_mld_version_cmd,
+      "ipv6 mld version (1-2)",
+      IPV6_STR
+      IFACE_MLD_STR
+      "MLD version\n"
+      "MLD version number\n")
+{
+	nb_cli_enqueue_change(vty, "./enable", NB_OP_MODIFY, "true");
+	/* TBD
+	 * nb_cli_enqueue_change(vty, "./mld-version", NB_OP_MODIFY,
+	 * argv[3]->arg); return nb_cli_apply_changes(vty,
+	 * FRR_GMP_INTERFACE_XPATH, "frr-routing:ipv6");
```

Comment:
In Yang, is this new field not defined?

Reply:
It is defined, but function is not implemented for mld so I have commented it